universal-application-tool-0.0.1/app/services/applicant/ReadOnlyApplicantProgramServiceImpl.java

```diff
@@ -19,6 +19,14 @@ protected ReadOnlyApplicantProgramServiceImpl(
     this.programDefinition = checkNotNull(programDefinition);
   }
 
+  /**
+   * Get the list of {@link Block}s this applicant should fill out for this program. This list
+   * includes any block that is incomplete or has errors (which indicate the applicant needs to make
```

Comment:
Thought (non-blocking, no recommended change): we don't save updates that result in errors, so we shouldn't ever have a block that is complete but also has errors

Reply:
Good point that we don't save error blocks - I'm wondering if this might come into play during updates still. We build a `ReadOnlyApplicantProgramService` based on the current block we're updating and the ApplicantData where we just staged updates, so that (I think) we can determine if the next block for the applicant is still the block they were working on if it has errors